FastSimulation/Tracking/src/SeedFinderSelector.cc

```diff
@@ -119,8 +153,108 @@ bool SeedFinderSelector::pass(const std::vector<const FastTrackerRecHit *>& hits
 	    multiHitGenerator_->hitTriplets(*trackingRegion_,tripletresult,*eventSetup_,result,&thmp,thirdLayerDetLayer,1);
 	    return !tripletresult.empty();
 	}
+	else if(CAHitTriplGenerator_)
+	{  
+	  return true;
+	}
+    }
 
+    if(CAHitQuadGenerator_)
+    {
+      std::cout<<"Inside CAHit section"<<std::endl;
+      if(hits.size() < 4)
+	{
+	  throw cms::Exception("FastSimTracking") << "For the given configuration, SeedFinderSelector::pass requires at least 4 hits";
+        }
+
+      if(!seedingLayer)
+	std::cout<<"SeedingLayers pointer not set"<<std::endl;
+
+      SeedingLayerSetsHits & layers = *seedingLayer;
+
+      for(int i=0; i<(int)hits.size()-1; i++){
```

Comment:
Shouldn't this loop be limited to `i < 3`?

Reply:
Correct... But the expression also gives 3 since hits.size() = 4 inside the quad section. But I guess it doesn't have to look so complicated when it is so obvious. So I have changed "(int)hits.size()-1" to 3.

Reply:
Ok, didn't think of that. In that case keeping the code but adding a comment explaining the situation would have been ok as well.